### PR TITLE
qa/cephfs: upgrade caps_helper.py

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
Related PR - PR https://github.com/ceph/ceph/pull/41779. Commits on this PR originally were located on PR https://github.com/ceph/ceph/pull/41779. These have been moved to a separate PR for ease of review, testing, merging and, more importantly to get these commits merged sooner so that this code can be reused to write other PRs.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
  - [x] QA Improvements
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by rishabh-d-dave** [comment]
jenkins test make check arm64

**Comment by rishabh-d-dave** [comment]
jenkins test make check

**Comment by rishabh-d-dave** [comment]
jenkins take make check

**Comment by rishabh-d-dave** [comment]
Tested all 5 PRs that were carved out of "fs authorize" PR (PR #41779). Testing was successful -

http://pulpito.front.sepia.ceph.com/rishabh-2022-07-07_17:45:28-fs-wip-rishabh-qa-cleanup-distro-default-smithi/
http://pulpito.front.sepia.ceph.com/rishabh-2022-07-07_17:19:32-fs-wip-rishabh-qa-cleanup-distro-default-smithi/

**Comment by rishabh-d-dave** [comment]
@vshankar First 2 commits on this PR are from PR #46993 and PR #46991.

**Comment by rishabh-d-dave** [comment]
jenkins test windows

**Comment by rishabh-d-dave** [comment]
This PR affects test_multifs_auth.py and test_admin.py. Both ran fine -

test_multifs_auth.py: http://pulpito.front.sepia.ceph.com/rishabh-2022-07-20_15:01:02-fs-wip-rishabh-qa-caps-helper-upgrade-distro-default-smithi/
test_admin.py: http://pulpito.front.sepia.ceph.com/rishabh-2022-07-20_15:01:19-fs-wip-rishabh-qa-caps-helper-upgrade-distro-default-smithi

File: qa/tasks/cephfs/caps_helper.py [hunk 1]
```diff
@@ -46,8 +74,6 @@ def conduct_pos_test_for_read_caps(self, filepaths, filedata, mounts):
                     self.assertEqual(data, contents)
 
     def conduct_pos_test_for_write_caps(self, filepaths, mounts):
-        filedata = ('some new data on first fs', 'some new data on second fs')
```
Comment (lxbsz) [inline qa/tasks/cephfs/caps_helper.py:49]: You remove the local `filedata` parameter in the first commit, but it's still used by the following code in this func. Maybe you have fixed in later commits, but it's hard to review for the code, could you fix it here ?

File: qa/tasks/cephfs/caps_helper.py [hunk 2]
```diff
@@ -1,80 +1,195 @@
...
+        log.info('testing for presence of these FS names in output of '
+                 '"fs ls" command run by client.')
+        for fsname in self._get_fsnames_from_moncap(moncap):
+            self.assertIn('name: ' + fsname, fsls)
 
-    def run_mds_cap_tests(self, filepaths, filedata, mounts, perm):
-        self.conduct_pos_test_for_read_caps(filepaths, filedata, mounts)
+    def run_mds_cap_tests(self, perm, mntpt=None):
+        """
+        Run test for read perm and run write perm, run positive test if it
```
Comment (lxbsz) [inline qa/tasks/cephfs/caps_helper.py:141]: I think `it` here means the `write perm`, right ?

File: qa/tasks/cephfs/test_admin.py [hunk 3]
```diff
@@ -1252,91 +1248,72 @@ def test_single_path_authorize_on_nonalphanumeric_fsname(self):
...
 
-    def setup_for_multiple_paths(self, perm, paths):
-        filedata, filename = 'some data on fs 1', 'file_on_fs1'
+    def setup_for_multiple_paths(self, fs_auth_caps):
+        for i, c in enumerate(fs_auth_caps):
+            PATH = c[0]
+            self.captesters[i].write_test_files((self.mount_a,), PATH)
 
-        self.mount_a.run_shell('mkdir -p dir1/dir12/dir13 dir2/dir22/dir23')
+        self.fs.authorize(self.client_id, fs_auth_caps)
```
Comment (lxbsz) [inline qa/tasks/cephfs/test_admin.py:1310]: Just a little curious why not do the remount in multiple paths case ?

Comment (lxbsz) [reply qa/tasks/cephfs/test_admin.py:1310]: Not sure whether will this hit the same issue in https://github.com/ceph/ceph/pull/44679/commits/1bd5beb2f1fb4fe587032542fff38fbc5febaef4 if do not remount it.

Comment (rishabh-d-dave) [reply qa/tasks/cephfs/test_admin.py:1310]: > Just a little curious why not do the remount in multiple paths case ?

CephFS is being re-mounted for the case of multiple paths too but in part where read-write perms are exercised and not in the setup part. The quote code above is setup part.

In that method, for each path, CephFS is remount with the path and then methods for exercising read-write perms in MON and MDS caps called.

File: qa/tasks/cephfs/caps_helper.py (hunk 2 above)
Comment (rishabh-d-dave) [reply qa/tasks/cephfs/caps_helper.py:141]: `and, for write perm, `

Comment (rishabh-d-dave) [reply qa/tasks/cephfs/caps_helper.py:141]: Fixed now.

File: qa/tasks/cephfs/caps_helper.py (hunk 1 above)
Comment (rishabh-d-dave) [reply qa/tasks/cephfs/caps_helper.py:49]: Yes, that would be better. Fixed now.

Comment (rishabh-d-dave) [reply qa/tasks/cephfs/caps_helper.py:49]: Moved here now - https://github.com/ceph/ceph/pull/46994/commits/370449ac9d088ba33111fb4948b1bf59f7901168

File: qa/tasks/cephfs/caps_helper.py (hunk 2 above)
Comment (rishabh-d-dave) [reply qa/tasks/cephfs/caps_helper.py]: https://github.com/ceph/ceph/pull/46994/commits/370449ac9d088ba33111fb4948b1bf59f7901168#diff-9998476d9d02f4136e9f6b4b512be93a77a5911a30027bec6a120706a1a024f3R88-R89

**Review by batrick — APPROVED** [review]
nit: `qa/cephfs: add a helper method in caps_helper.py` contains an unrelated method, seemingly (`run_cap_tests`)

Looks fine otherwise if your testing is good. I did not go through it in depth.

**Review by lxbsz — APPROVED** [review]

**Comment by rishabh-d-dave** [comment]
QA was successful - https://tracker.ceph.com/projects/cephfs/wiki/Main#2022-Aug-26.

**Comment by rishabh-d-dave** [comment]
`test_admin.py` was not executed during QA run. I ran it separately - http://pulpito.front.sepia.ceph.com/rishabh-2022-08-26_11:25:16-fs-wip-rishabh-testing-2022Aug19-distro-default-smithi/.

**Pull request merged**